pkg/project/v2/project_loader.go

```diff
@@ -84,13 +83,95 @@ func LoadProjects(fs afero.Fs, context ProjectLoaderContext) ([]Project, []error
 		return nil, []error{fmt.Errorf("no projects defined in manifest")}
 	}
 
-	log.Info("Loading %d projects...", len(context.Manifest.Projects))
+	if len(specificProjects) == 0 {
+		log.Info("Loading %d projects...", len(context.Manifest.Projects))
+		return loadProjectsFromProjectDefinitions(workingDirFs, context, context.Manifest.Projects, environments)
+	}
 
-	var errors []error
+	specificProjectDefinitions, err := filterProjectDefinitionsByProjectNames(context.Manifest.Projects, specificProjects)
+	if err != nil {
+		return nil, []error{err}
+	}
+	log.Info("Loading %d projects...", len(specificProjectDefinitions))
+	projects, errors := loadProjectsFromProjectDefinitions(workingDirFs, context, specificProjectDefinitions, environments)
+	if errors != nil {
+		return nil, errors
+	}
 
-	for _, projectDefinition := range context.Manifest.Projects {
-		project, projectErrors := loadProject(workingDirFs, context, projectDefinition, environments)
+	additionalDepedencyProjectNames := getAdditionalDependencyProjectNames(projects, environments)
+	if len(additionalDepedencyProjectNames) > 0 {
```

Comment:
Is this only 1 pass?

What about dependencies that are layered 3 and deeper?
`A --dependsOn--> B --dependsOn--> C`

`Load A` should load `C` as well, but if I see it correctly, you do only one pass (?)

Reply:
Very good point, I will test this and see - but I think you're right!

Reply:
Now fixed, together with some additional tests